django/contrib/auth/admin.py

```diff
@@ -205,6 +205,7 @@ def response_add(self, request, obj, post_url_continue=None):
         # * The user has pressed the 'Save and add another' button
         # * We are adding a user in a popup
         if '_addanother' not in request.POST and IS_POPUP_VAR not in request.POST:
+            request.POST = request.POST.copy()
```

Comment:
Is there any more graceful way we could do this? It seems odd to make `request.POST` immutable, but then to get bpast that by replacing it with a mutable instance in this cases. (May not be a show stopper, but worth considering)